app/src/main/java/org/mozilla/fenix/library/history/HistoryController.kt

```diff
@@ -145,14 +150,19 @@ class DefaultHistoryController(
         scope.launch {
             store.dispatch(HistoryFragmentAction.EnterDeletionMode)
             if (timeFrame == null) {
-                GleanHistory.removedAll.record(mozilla.telemetry.glean.private.NoExtras())
                 historyStorage.deleteEverything()
             } else {
+                val longRange = timeFrame.toLongRange()
                 historyStorage.deleteVisitsBetween(
-                    startTime = timeFrame.toLongRange().first,
-                    endTime = timeFrame.toLongRange().last,
+                    startTime = longRange.first,
+                    endTime = longRange.last,
                 )
             }
+            when (timeFrame) {
+                RemoveTimeFrame.LastHour -> GleanHistory.removedLastHour.record(NoExtras())
+                RemoveTimeFrame.TodayAndYesterday -> GleanHistory.removedTodayAndYesterday.record(NoExtras())
+                null -> GleanHistory.removedAll.record(NoExtras())
```

Comment:
I know this is not in the scope of this PR, but maybe `RemoveTimeFrame` should also have an `All` enum value, instead of passing null in all places.

Reply:
What value would it have? 
I was thinking about that as well, but`RemoveTimeFrame` is basically a utility class for `PlacesHistoryStorage`. It doesn't use any values when `deleteEverything` is called. So if I create `RemoveTimeFrame.ALL` with something like 0 to max_value it's values would never be used.

Reply:
created a follow up [issue](https://github.com/mozilla-mobile/fenix/issues/26179)
would appreciate any ideas on implementation👾

Reply:
Let's not do that change in this PR.  There is a high probability that once QA verifies this change we will be lifting this to the release branch.  It's safer if we have the smallest change possible.